photon-ml/src/main/scala/com/linkedin/photon/ml/Params.scala

```diff
@@ -158,6 +160,9 @@ class Params {
    */
   var deleteOutputDirsIfExist: Boolean = false
 
+  var inputFormatType: InputFormatType = InputFormatType.AVRO
+  var featureDimension: Int = -1
```

Comment:
We need ```featureDimension``` here because it is a required parameter to construct a sparse vector of type ```breeze.linalg.SparseVector```.

However, I would prefer us not exposing such implementation details to the users. Is it possible to let the index map loaders to figure out the total number of features instead of asking the users to specify it explicitly? 

For example, instead of having ```featureDimension``` to be a constructor parameter of ```IdentityIndexMapLoader```, maybe we can have factory method similar to ```createDefaultIndexMapLoader``` that figures out the number of features in the data set automatically.